File: ncPersistencia/pCheque.vb
Imports ncDados.nsCheques
Imports ncComum.nsAcessoBD
Imports ncComum.nsFuncoes
Imports ncComum.nsExcecao

Namespace nsCheques

  Public Class pCheques

    Public Function Listar() As ColecaoCheques

      Dim retorno As ColecaoCheques
      Dim acessoBanco As cAcessoBD
      Dim ds As DataSet
      Dim dt As DataTable
      Dim row As DataRow
      Dim item As dCheques
      Dim comandoSQL As String


      Try

        acessoBanco = New cAcessoBD


        comandoSQL = " Select cid, " & _
                     " dataemissao, datadeposito, valor, numero, vendaid, clienteid, baixado, " & _
                     " bancoCodigo, bancoNome, agencia, conta From cheques"

        ds = acessoBanco.ExecutarDS(comandoSQL)

        If Not ds Is Nothing Then
          If ds.Tables.Count > 0 Then
            dt = ds.Tables(0)

            If dt.Rows.Count > 0 Then
              retorno = New ColecaoCheques

              For Each row In dt.Rows
                item = New dCheques

                item.cid = cFuncoes.RetornarInteiro(row("cid"))
                item.dataEmissao = cFuncoes.RetornarData(row("dataemissao"))
                item.dataDeposito = cFuncoes.RetornarData(row("datadeposito"))
                item.valor = cFuncoes.RetornarDecimal(row("valor"))
                item.Numero = cFuncoes.RetornarTexto(row("numero"))
                item.vendasId = cFuncoes.RetornarInteiro(row("vendasid"))
                item.clienteId = cFuncoes.RetornarInteiro(row("clienteid"))
                item.baixado = cFuncoes.RetornarTexto(row("baixado"))
                item.bancoCodigo = cFuncoes.RetornarTexto(row("bancoCodigo"))
                item.bancoNome = cFuncoes.RetornarTexto(row("bancoNome"))
                item.agencia = cFuncoes.RetornarTexto(row("agencia"))
                item.conta = cFuncoes.RetornarTexto(row("conta"))

                retorno.Add(item)
              Next
            Else
              retorno = Nothing
            End If
          Else
            retorno = Nothing
          End If
        Else
          retorno = Nothing
        End If

      Catch ex As Exception

        retorno = Nothing
        Throw New ExcecaoNascomercio("Erro em Listar Cheques [" & Me.ToString() & "] - " & ex.Message)

      End Try

      Listar = retorno

    End Function

    Public Function Consultar(ByVal dados As dCheques) As ColecaoCheques

      Dim retorno As ColecaoCheques
      Dim acessoBanco As cAcessoBD
      Dim ds As DataSet
      Dim dt As DataTable
      Dim row As DataRow
      Dim item As dCheques
      Dim sqlSelect As String
      Dim sqlWhere As String
      Dim sqlFrom As String


      Try

        acessoBanco = New cAcessoBD


        sqlSelect = " Select cid, " & _
                    " dataemissao, datadeposito, valor, numero, vendaid, clienteid, baixado, " & _
                    " bancoCodigo, bancoNome, agencia, conta "

        sqlWhere = String.Empty
        sqlFrom = " From cheques "

        '-- cid
        sqlWhere = cFuncoes.MontarParametrosSQL(sqlWhere, dados.cid, "cid")

        '-- Numero
        If dados.Numero <> 0 Then
          sqlWhere = cFuncoes.MontarParametrosSQL(sqlWhere, dados.Numero, "numero")
        End If

        If Not sqlWhere.Equals(String.Empty) Then
          sqlWhere = " WHERE " & sqlWhere
        End If

        ds = acessoBanco.ExecutarDS(sqlSelect & " " & sqlFrom & " " & sqlWhere)

        If Not ds Is Nothing Then
          If ds.Tables.Count > 0 Then
            dt = ds.Tables(0)

            If dt.Rows.Count > 0 Then
              retorno = New ColecaoCheques

              For Each row In dt.Rows
                item = New dCheques

                item.cid = cFuncoes.RetornarInteiro(row("cid"))
                item.dataEmissao = cFuncoes.RetornarData(row("dataemissao"))
                item.dataDeposito = cFuncoes.RetornarData(row("datadeposito"))
                item.Valor = cFuncoes.RetornarDecimal(row("valor"))
                item.Numero = cFuncoes.RetornarTexto(row("numero"))
                item.vendasId = cFuncoes.RetornarInteiro(row("vendaid"))
                item.clienteId = cFuncoes.RetornarInteiro(row("clienteid"))
                item.baixado = cFuncoes.RetornarTexto(row("baixado"))
                item.bancoCodigo = cFuncoes.RetornarTexto(row("bancoCodigo"))
                item.bancoNome = cFuncoes.RetornarTexto(row("bancoNome"))
                item.agencia = cFuncoes.RetornarTexto(row("agencia"))
                item.conta = cFuncoes.RetornarTexto(row("conta"))

                retorno.Add(item)
              Next
            Else
              retorno = Nothing
            End If
          Else
            retorno = Nothing
          End If
        Else
          retorno = Nothing
        End If

      Catch ex As Exception

        retorno = Nothing
        Throw New ExcecaoNascomercio("Erro em Consultar Cheques[" & Me.ToString() & "] - " & ex.Message)

      End Try

      Consultar = retorno

    End Function

    Public Function Incluir(ByVal dados As dCheques) As Integer

      Dim retorno As Integer
      Dim acessoBanco As cAcessoBD
      Dim comandoSQL As String


      Try

        acessoBanco = New cAcessoBD

        comandoSQL = " INSERT INTO " & _
            " cheques ( dataemissao, datadeposito, valor, numero, vendaid, clienteid, baixado, " & _
            " bancoCodigo, bancoNome, agencia, conta ) " & _
            " VALUES (" & _
                    cFuncoes.PersistirData(dados.dataEmissao) & "," & _
                    cFuncoes.PersistirData(dados.dataDeposito) & "," & _
                    cFuncoes.PersistirDecimal(dados.valor) & "," & _
                    cFuncoes.PersistirTexto(dados.Numero) & "," & _
                    cFuncoes.PersistirInteiro(dados.vendasId) & "," & _
                    cFuncoes.PersistirInteiro(dados.clienteId) & "," & _
                    cFuncoes.PersistirTexto(dados.baixado) & "," & _
                    cFuncoes.PersistirTexto(dados.bancoCodigo) & "," & _
                    cFuncoes.PersistirTexto(dados.bancoNome) & "," & _
                    cFuncoes.PersistirTexto(dados.agencia) & "," & _
                    cFuncoes.PersistirTexto(dados.conta) & ")"

        retorno = acessoBanco.ExecutarCID(comandoSQL)

      Catch ex As Exception

        retorno = Nothing
        Throw New ExcecaoNascomercio("Erro em Incluir Cheques [" & Me.ToString() & "] - " & ex.Message)

      End Try

      Incluir = retorno

    End Function

    Public Function Alterar(ByVal dados As dCheques) As Integer

      Dim retorno As Integer
      Dim acessoBanco As cAcessoBD
      Dim comandoSQL As String

      Try

        acessoBanco = New cAcessoBD

        comandoSQL = " UPDATE cheques SET " & _
            " dataEmissao = " & cFuncoes.PersistirData(dados.dataEmissao) & "," & _
            " dataDeposito = " & cFuncoes.PersistirData(dados.dataDeposito) & "," & _
            " valor = " & cFuncoes.PersistirDecimal(dados.valor) & "," & _
            " numero = " & cFuncoes.PersistirTexto(dados.Numero) & "," & _
            " vendaid = " & cFuncoes.PersistirInteiro(dados.vendasId) & "," & _
            " clienteid = " & cFuncoes.PersistirInteiro(dados.clienteId) & "," & _
            " baixado = " & cFuncoes.PersistirTexto(dados.baixado) & "," & _
            " bancoCodigo = " & cFuncoes.PersistirTexto(dados.bancoCodigo) & "," & _
            " bancoNome = " & cFuncoes.PersistirTexto(dados.bancoNome) & "," & _
            " agencia = " & cFuncoes.PersistirTexto(dados.agencia) & "," & _
            " conta = " & cFuncoes.PersistirTexto(dados.conta) & _
            " WHERE " & _
            " cid = " & dados.cid.ToString()

        retorno = acessoBanco.ExecutarINT(comandoSQL)

      Catch ex As Exception

        retorno = Nothing
        Throw New ExcecaoNascomercio("Erro em Alterar Cheque [" & Me.ToString() & "] - " & ex.Message)

      End Try

      Alterar = retorno

    End Function

    Public Function Baixar() As Integer

      Dim retorno As Integer
      Dim acessoBanco As cAcessoBD
      Dim comandoSQL As String

      Try

        acessoBanco = New cAcessoBD

        comandoSQL = " UPDATE cheques SET " & _
            " baixado = 'Sim'" & _
            " WHERE (baixado <> 'Sim') AND " & _
            " dataDeposito < '" & Today.ToString("yyyy-MM-dd") & "'"

        retorno = acessoBanco.ExecutarINT(comandoSQL)

      Catch ex As Exception

        retorno = Nothing
        Throw New ExcecaoNascomercio("Erro em Alterar Cheque [" & Me.ToString() & "] - " & ex.Message)

      End Try

      Return retorno

    End Function
    Public Function Excluir(ByVal dados As dCheques) As Integer

      Dim retorno As Integer
      Dim acessoBanco As cAcessoBD
      Dim comandoSQL As String

      Try

        acessoBanco = New cAcessoBD

        comandoSQL = " DELETE FROM cheques " & _
            " WHERE cid = " & dados.cid.ToString()

        retorno = acessoBanco.ExecutarINT(comandoSQL)

      Catch ex As Exception

        retorno = Nothing
        Throw New ExcecaoNascomercio("Erro em Excluir Cheque [" & Me.ToString() & "] - " & ex.Message)

      End Try

      Excluir = retorno

    End Function

  End Class

End Namespace

File: ncRepositorios/pCheque.cs
using System;
using System.Collections.Generic;
using Dapper;
using MySql.Data.MySqlClient;
using ncNComum.nsExcecao;
using nsCheques;

namespace ncPersistencia.nsCheques
{
    public class pCheques : RepositorioBase, IpCheques
    {
        public ColecaoCheques Listar()
        {
            try
            {
                using (var conn = CriarConexao())
                {
                    var lista = conn.Query<dCheques>(
                        "SELECT cid, emitente, valor, dataEmissao, dataDeposito, banco, agencia, conta, numero, baixado FROM cheques").AsList();
                    if (lista.Count == 0) return null;
                    var retorno = new ColecaoCheques();
                    retorno.AddRange(lista);
                    return retorno;
                }
            }
            catch (ExcecaoNascomercio) { throw; }
            catch (Exception ex)
            {
                throw new ExcecaoNascomercio("Erro em Listar Cheques [" + ToString() + "] - " + ex.Message);
            }
        }

        public ColecaoCheques Consultar(dCheques dados)
        {
            try
            {
                using (var conn = CriarConexao())
                {
                    var conditions = new List<string>();
                    var p = new DynamicParameters();
                    if (dados.cid != null && dados.cid != 0) { conditions.Add("cid=@cid"); p.Add("cid", dados.cid); }
                    if (dados.valor != null && dados.valor != 0) { conditions.Add("valor=@valor"); p.Add("valor", dados.valor); }
                    if (dados.dataEmissao != default(DateTime)) { conditions.Add("dataEmissao=@dataEmissao"); p.Add("dataEmissao", dados.dataEmissao); }
                    if (dados.dataDeposito != default(DateTime)) { conditions.Add("dataDeposito=@dataDeposito"); p.Add("dataDeposito", dados.dataDeposito); }
                    if (!string.IsNullOrEmpty(dados.bancoCodigo)) { conditions.Add("banco=@banco"); p.Add("banco", dados.bancoCodigo); }
                    if (!string.IsNullOrEmpty(dados.agencia)) { conditions.Add("agencia=@agencia"); p.Add("agencia", dados.agencia); }
                    if (!string.IsNullOrEmpty(dados.conta)) { conditions.Add("conta=@conta"); p.Add("conta", dados.conta); }
                    if (!string.IsNullOrEmpty(dados.Numero)) { conditions.Add("numero=@numero"); p.Add("numero", dados.Numero); }
                    if (!string.IsNullOrEmpty(dados.baixado)) { conditions.Add("baixado=@baixado"); p.Add("baixado", dados.baixado); }
                    var where = conditions.Count > 0 ? "WHERE " + string.Join(" AND ", conditions) : "";
                    var sql = $"SELECT cid, emitente, valor, dataEmissao, dataDeposito, banco, agencia, conta, numero, baixado FROM cheques {where}";
                    var lista = conn.Query<dCheques>(sql, p).AsList();
                    if (lista.Count == 0) return null;
                    var retorno = new ColecaoCheques();
                    retorno.AddRange(lista);
                    return retorno;
                }
            }
            catch (ExcecaoNascomercio) { throw; }
            catch (Exception ex)
            {
                throw new ExcecaoNascomercio("Erro em Consultar Cheques [" + ToString() + "] - " + ex.Message);
            }
        }

        public int Incluir(dCheques dados)
        {
            try
            {
                using (var conn = CriarConexao())
                {
                    conn.Execute(
                        "INSERT INTO cheques (emitente, valor, dataEmissao, dataDeposito, banco, agencia, conta, numero, baixado) " +
                        "VALUES (@emitente, @valor, @dataEmissao, @dataDeposito, @banco, @agencia, @conta, @numero, @baixado)",
                        new {
                            emitente = dados.cid, valor = dados.valor, dataEmissao = dados.dataEmissao,
                            dataDeposito = dados.dataDeposito, banco = dados.bancoCodigo, agencia = dados.agencia,
                            conta = dados.conta, numero = dados.Numero, baixado = dados.baixado
                        });
                    return (int)conn.ExecuteScalar<long>("SELECT LAST_INSERT_ID()");
                }
            }
            catch (ExcecaoNascomercio) { throw; }
            catch (Exception ex)
            {
                throw new ExcecaoNascomercio("Erro em Incluir Cheques [" + ToString() + "] - " + ex.Message);
            }
        }

        public int Alterar(dCheques dados)
        {
            try
            {
                using (var conn = CriarConexao())
                {
                    return conn.Execute(
                        "UPDATE cheques SET emitente=@emitente, valor=@valor, dataEmissao=@dataEmissao, dataDeposito=@dataDeposito, " +
                        "banco=@banco, agencia=@agencia, conta=@conta, numero=@numero, baixado=@baixado WHERE cid=@cid",
                        new {
                            emitente = dados.cid, valor = dados.valor, dataEmissao = dados.dataEmissao,
                            dataDeposito = dados.dataDeposito, banco = dados.bancoCodigo, agencia = dados.agencia,
                            conta = dados.conta, numero = dados.Numero, baixado = dados.baixado, cid = dados.cid
                        });
                }
            }
            catch (ExcecaoNascomercio) { throw; }
            catch (Exception ex)
            {
                throw new ExcecaoNascomercio("Erro em Alterar Cheques [" + ToString() + "] - " + ex.Message);
            }
        }

        public int Baixar()
        {
            try
            {
                using (var conn = CriarConexao())
                {
                    return conn.Execute("UPDATE cheques SET baixado='Sim' WHERE dataDeposito < date(now())");
                }
            }
            catch (ExcecaoNascomercio) { throw; }
            catch (Exception ex)
            {
                throw new ExcecaoNascomercio("Erro em Baixar Cheques [" + ToString() + "] - " + ex.Message);
            }
        }

        public int Excluir(dCheques dados)
        {
            try
            {
                using (var conn = CriarConexao())
                {
                    return conn.Execute("DELETE FROM cheques WHERE cid=@cid", new { cid = dados.cid });
                }
            }
            catch (ExcecaoNascomercio) { throw; }
            catch (Exception ex)
            {
                throw new ExcecaoNascomercio("Erro em Excluir Cheques [" + ToString() + "] - " + ex.Message);
            }
        }
    }
}
